### ISSUE TITLE
Simplify install target config requirements?

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#5261]
With scheduler and job hosts on the same FS, all platform configs  - including `hosts = localhost` or `hosts = single-host` result in job init failure if I do not explicitly define an `install target` (due to unexpected files found on disk).

Given that job platform hosts have to be on a shared FS, why do we have to explicitly define an install target when (presumably) any one of the hosts, and often localhost even if not a platform host, would do?

Can we not just do this?:
- choose a random install target from the list of hosts
- and handle the high likelihood that localhost (i.e. scheduler run host) is on the same FS, but not a job host, by writing a unique run-specific UUID to the workflow run directory and check for that when trying the randomly chosen install target (if the UUID is found, no need to install files there).
- I think the UUID would also handle use of multiple platforms on the same FS, in the same workflow

I went back to the platforms proposal, but it doesn't seem to give an obvious reason why install targets needs to explicitly defined (but maybe I'm missing something!).

**Comment by dpmatthews** [comment]
It was designed this way in order to prevent unnecessary ssh's. Better to get the config right rather than attempt unnecessary remote init's every time a workflow is run. You could also get in a mess if trying to init the same install target multiple times in parallel, especially give that remote file installation can take a long time (doesn't apply to localhost but does to the more general case).

There's also a complication with symlinks since these are configured per install target.

Also, some operations, such as remote tidy, are performed per install target.

**Comment by hjoliver** [comment]
OK, I see the "avoid unnecessary ssh" reason, not sure if the others are valid (e.g. wouldn't remote tidy work the same way as remote install?).

In that case, can we raise an exception on parsing the global config if a platform does not define an install target, rather than wait till job init at run time for the inevitable failure?

**Comment by dpmatthews** [comment]
The install target defaults to the platform name so we can't raise an exception.

We have thought about writing a utility to check your platform setup (check all hosts are reachable, check if any install targets share the same filesystem).

**Comment by hjoliver** [comment]
> The install target defaults to the platform name so we can't raise an exception.

That now strikes me as unhelpful, given that (multi-host) platforms will not use a host name as the platform name, right?  Is that designed to handle single host platforms?   If so, what about raising an error if a multi-host platform does not define an install target?  (And/or if the platform name is not a known host name).

**Comment by dpmatthews** [comment]
This is a perfectly reasonable config where it shouldn't be necessary to define the install target:
```
    [[hpc]]
        # install target = hpc
        hosts = login1, login2
        job runner = pbs
```

**Comment by hjoliver** [comment]
But what is "hpc" in that example?  An alias for one of the login nodes? If it is, then that's fine, but we could test that to make sure it is and therefore distinguish that valid config from this broken one:

```
[[hpc_pbs]]
   hosts = login1, login2
   job runner = pbs
```
... which looks just as reasonable, but will fail at job init.

**Comment by dpmatthews** [comment]
I'm confused. "hpc" was the just platform name in my example. In your example you've changed the name to "hpc_pbs" but that doesn't really change anything - neither are broken as far as I can see.

**Comment by hjoliver** [comment]
You said:

>         # install target = hpc

Which I took to mean that "hpc" is literally a valid host name.  And you said that install target defaults to platform name.

So, by "hpc_pbs", I meant a descriptive platform name that is not a valid host name in itself, so the install target defaulting to that won't do any good.

According to my tests, these examples are broken if the scheduler run host is on the same shared filesystem as the job hosts, because no install target is declared, and job init fails due to workflow files written on the scheduler run host.

**Comment by dpmatthews** [comment]
> According to my tests, these examples are broken if the scheduler run host is on the same shared filesystem as the job hosts, because no install target is declared, and job init fails due to workflow files written on the scheduler run host.

Yes, that's true. Any platform that that shares it's filesystems with the scheduler host needs to include
```
install target = localhost
```
However, if the jobs hosts do not share filesystems with the scheduler host both examples are fine.

> by "hpc_pbs", I meant a descriptive platform name that is not a valid host name in itself, so the install target defaulting to that won't do any good

"install target" is just a label - it's irrelevant whether it happens to be a valid host name.
All that matters is that platforms which share the same filesystems use the same label.
So, "hpc_pbs" is fine as the default for "install target" (assuming it doesn't share filesystems with the scheduler host).
